speechbrain/core.py

```diff
@@ -928,8 +937,13 @@ def fit_batch(self, batch):
                 self.zero_grad()
                 self.optimizer_step += 1
         else:
-            outputs = self.compute_forward(batch, Stage.TRAIN)
-            loss = self.compute_objectives(outputs, batch, Stage.TRAIN)
+            if self.bfloat16_mix_prec:
```

Comment:
I think something like this is a good idea. This is not a necessary change, but a slightly more succinct way to write this would be something like:

```
with torch.cuda.amp.autocast(dtype=torch.bfloat16, enabled=self.bfloat16_mix_prec):
    outputs = self.compute_forward(batch, Stage.TRAIN)
   loss = self.compute_objectives(outputs, batch, Stage.TRAIN)
```

Reply:
Yeah true, I'll fix this.

Reply:
Done.